src/interface/block_chain.cpp

```diff
@@ -293,6 +299,251 @@ void block_chain::push(transaction_const_ptr tx, dispatcher&,
     handler(database_.push(*tx, chain_state()->enabled_forks()));
 }
 
+
+//Mark every previous output of the transaction as TEMPORARY SPENT
+void block_chain::append_spend(transaction_const_ptr tx) {
+    std::vector<prev_output> prev_outputs;
+    for (auto const& input : tx->inputs()) {
+        auto const& output_point = input.previous_output();
+        prev_outputs.push_back({output_point.hash(), output_point.index()});
+    }
+    chosen_spent_.insert({tx->hash(), std::move(prev_outputs)});
+
+}
+
+// Once a transaction was mined, every previous output that was marked
+// as TEMPORARY SPENT, needs to be removed
+// (since it was permanetly added to the spent database)
+void block_chain::remove_spend(libbitcoin::hash_digest hash){
+    chosen_spent_.erase(hash);
+}
+
+// Search the TEMPORARY SPENT index for conflicts between previous output
+std::set<libbitcoin::hash_digest> block_chain::get_double_spend_chosen_list(transaction_const_ptr tx) {
+
+    std::set<libbitcoin::hash_digest> spent_conflict{};
+    for (auto const& input : tx->inputs()) {
+        for(auto const& spents : chosen_spent_){
+            for(auto const& previous : spents.second){
+                if(previous.output_hash == input.previous_output().hash() &&
+                    previous.output_index == input.previous_output().index())
+                {
+                    spent_conflict.insert(tx->hash());
+                }
+            }
+        }
+    }
+    return spent_conflict;
+}
+
+// Search the spent database and the TEMPORARY SPENT for conflicts
+bool block_chain::check_is_double_spend(transaction_const_ptr tx){
+    bool is_double_spend = false;
+    for(auto const& input : tx->inputs()){
+        boost::latch latch(2);
+        fetch_spend(input.previous_output(), [&](const libbitcoin::code &ec, libbitcoin::chain::input_point input) {
```

Comment:
It is convenient to create a synchronic get_spend() function to make the code more readable